apps/api/src/jobs/consumers/notifications/waTwilioBusiness-notifications.job.consumer.ts

```diff
@@ -32,4 +34,28 @@ export class WaTwilioBusinessNotificationsConsumer extends NotificationConsumer
       );
     });
   }
+
+  async processWaTwilioBusinessNotificationConfirmationQueue(id: number): Promise<void> {
+    return super.processAwaitingConfirmationNotificationQueue(id, async () => {
+      const notification = (await this.notificationsService.getNotificationById(id))[0];
+      const result = await this.waTwilioBusinessService.getDeliveryStatus(
+        (notification.result.result as WaTwilioBusinessResponseData).sid as string,
+        notification.providerId,
+      );
+      const deliveryStatus = result.status;
+
+      if (ProviderDeliveryStatus.WA_TWILIO_BUSINESS.FAILURE_STATES.includes(deliveryStatus)) {
+        return { result, deliveryStatus: DeliveryStatus.PENDING };
```

Comment:
where did we add a check to keep in failed state if max retry is exhausted?

Reply:
The retryCount gets incremented in super.processAwaitingConfirmationNotificationQueue(). And since we will then attempt to resend the notification, the retry and failure marking is handled in NotificationConsumer's processNotificationQueue().